src/data/int/basic.lean

```diff
@@ -15,6 +15,9 @@ meta instance : has_to_format ℤ := ⟨λ z, int.rec_on z (λ k, ↑k) (λ k, "
 
 attribute [simp] int.coe_nat_add int.coe_nat_mul int.coe_nat_zero int.coe_nat_one int.coe_nat_succ
 
+@[simp] theorem add_eq_plus {a b : ℤ} : int.add a b = a + b := rfl
```

Comment:
I'm a little skeptical of these simp lemmas. They create a cycle if you try to unfold the `+` notation using `simp [(+)]`, or (I think) even `dsimp [(+)]`. These are pretty common idioms. You could argue that one should never unfold this notation, or use `simp only` instead, but people will do it by accident and the looping behavior will lead to confusion. @digama0 @johoelzl do you guys think these are reasonable simp rules?

Reply:
I'm skeptical too, but the alternative seems to make this tactic almost unusable --- lots of `nat.add`s in the output, which have to be manually converted back to the right form. Mario's suggestion for the `with [2,3,5,7]` syntax partly fixes this, but I feel like it would be a shame to essentially make this syntax compulsory.

Reply:
You have to manually call `simp` anyway, so you could just call it with this lemma as an argument. Or, the tactic could automatically try `simp only [add_eq_plus, ...]`.

I don't think `add_eq_plus` is the right name, by the way -- we generally don't use "plus" for addition. `int_add_eq_add`? `int.add_def`?

Reply:
I'm in favor of this rule as a `@[simp]` rule. We have the problem that `case` generates the wrong terms and we want to normalize it again. Then `simp` can operate further. `int.add`, `nat.add` etc are internals users shouldn't see and not need.

Reply:
I'll fix the name, and tentatively continue with this is a simp lemma per @johoelzl's comment.